cli: log more error context on startup

spilchen opened the pull request:
When checking for CLI errors, we now use `%+v` when logging to include additional context.  For example, if the error wraps a stack, the stack trace will now be part of the log output.  This provides better debugging information, particularly for issues like #135273.

Closes #142627

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142738)
<!-- Reviewable:end -->

rafiss reviewed (changes requested):
thanks! just thought of one additional check we might want here.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142738#-:-OL9y7XR-PibqM5chQCs:b-khudhv)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/cli/clierror/check.go` line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/142738#-OL9xluYCrsPaA5pc4Qc:-OL9xluYCrsPaA5pc4Qd:b-5xk4c5) ([raw file](https://github.com/cockroachdb/cockroach/blob/f98f783ad4a89d1f67773a4beeee3e5c9c755909/pkg/cli/clierror/check.go#L30)):*
> ```Go
> 		cause = ec.Unwrap()
> 	}
> 	logger(context.Background(), severity, "%+v", cause)
> ```

ah i didn't realize we'd need to change the logging in a production code path. always showing the stack trace may be alarming for some "expected" errors that might happen during startup (like incorrectly configured flags or something).

could we make this only use `%+v` for assertion errors? i believe the linked test encountered an assertion error, so that should help.


<!-- Sent from Reviewable.io -->

spilchen committed:
cli: log more error context on startup

When checking for CLI errors, we now use `%+v` when logging to include
additional context.  For example, if the error wraps a stack, the stack
trace will now be part of the log output.  This provides better
debugging information, particularly for issues like #135273.

Closes #142627

Release note: none

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142738)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/cli/clierror/check.go` line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/142738#-OL9xluYCrsPaA5pc4Qc:-OLATLkQA9LZ3PHGo5bJ:bggeek3) ([raw file](https://github.com/cockroachdb/cockroach/blob/f98f783ad4a89d1f67773a4beeee3e5c9c755909/pkg/cli/clierror/check.go#L30)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

ah i didn't realize we'd need to change the logging in a production code path. always showing the stack trace may be alarming for some "expected" errors that might happen during startup (like incorrectly configured flags or something).

could we make this only use `%+v` for assertion errors? i believe the linked test encountered an assertion error, so that should help.
</blockquote></details>

Good point. Yes, I'll add the additional context for assertion errors only.


<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142738)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/cli/clierror/check.go` line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/142738#-OL9xluYCrsPaA5pc4Qc:-OLAkxaQ9z6CcPly0Vi9:bnyegco) ([raw file](https://github.com/cockroachdb/cockroach/blob/f98f783ad4a89d1f67773a4beeee3e5c9c755909/pkg/cli/clierror/check.go#L30)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

Good point. Yes, I'll add the additional context for assertion errors only.
</blockquote></details>

the linter may want us to use a statically defined string for the format argument, but i'm not sure. if it doesn't, this lgtm; and if it does, it should be easy to resolve by doing the call to `logger` in the if-else.


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142738)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/cli/clierror/check.go` line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/142738#-OL9xluYCrsPaA5pc4Qc:-OLEBhgOBqvRMtyyhTJj:btoy0yw) ([raw file](https://github.com/cockroachdb/cockroach/blob/f98f783ad4a89d1f67773a4beeee3e5c9c755909/pkg/cli/clierror/check.go#L30)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

the linter may want us to use a statically defined string for the format argument, but i'm not sure. if it doesn't, this lgtm; and if it does, it should be easy to resolve by doing the call to `logger` in the if-else.
</blockquote></details>

The linter seems to be okay with the current change. I'm going to merge. Thanks.


<!-- Sent from Reviewable.io -->
